packages/ui/react-ui/src/hooks/useCrossMintModal.ts

```diff
@@ -0,0 +1,145 @@
+    collectionTitle?: string;
+    collectionDescription?: string;
+    collectionPhoto?: string;
+    mintTo?: string;
+    emailTo?: string;
+    listingId?: string;
+}
+
+const PROD_URL = "https://www.crossmint.io";
+const DEV_URL = "http://localhost:3001";
+
+const executeIfExists = (fn?: () => any | undefined):void => {
+    if (fn && typeof fn === 'function') fn();
+}
+
+const overlayId = '__crossmint-overlay__';
+
+const addLoadingOverlay = ():void => {
+    const overlayEl = document.createElement('div');
+    overlayEl.setAttribute('id', overlayId);
+    const overlayStyles = {
+        width: '100vw',
+        height: '100vh',
+        'background-color': 'rgba(0, 0, 0, 0.5)',
+        position: 'fixed',
+        'z-index': '99999999',
+        top: '0',
+        left: '0',
+    };
+    Object.assign(overlayEl.style, overlayStyles);
+    document.body.appendChild(overlayEl);
+}
+
+const removeLoadingOverlay = ():void => {
+    document.getElementById(overlayId)?.remove();
+}
+
+export default function useCrossMintModal({ development, clientId, crossmintOpened, crossmintClosed, showOverlay }: IProps): IReturn {
+    const [connecting, setConnecting] = useState(false);
+
+    const createPopup = (
+        collectionTitle?: string,
+        collectionDescription?: string,
+        collectionPhoto?: string,
+        mintTo?: string,
+        emailTo?: string,
+        listingId?: string
+    ) => {
+        const urlOrigin = development ? DEV_URL : PROD_URL;
+        const getMintQueryParams = ():string => {
+            const mintQueryParams:MintQueryParams = {
+                clientId: encodeURIComponent(clientId),
+                closeOnSuccess: 'false',
+            };
+
+            if (collectionTitle) mintQueryParams.collectionTitle = encodeURIComponent(collectionTitle);
+            if (collectionDescription) mintQueryParams.collectionDescription = encodeURIComponent(collectionDescription);
+            if (collectionPhoto) mintQueryParams.collectionPhoto = encodeURIComponent(collectionPhoto);
+            if (mintTo) mintQueryParams.mintTo = encodeURIComponent(mintTo);
+            if (emailTo) mintQueryParams.emailTo = encodeURIComponent(emailTo);
+            if (listingId) mintQueryParams.listingId = encodeURIComponent(listingId);
+
+            return new URLSearchParams(mintQueryParams).toString()
+        };
+        const callbackUrl = encodeURIComponent(`${urlOrigin}/checkout/mint?${getMintQueryParams()}`);
+        const url = `${urlOrigin}/signin?callbackUrl=${callbackUrl}`;
+
+        const pop = window.open(
+            url,
+            "popUpWindow",
+            createPopupString()
+        );
+        if (pop) {
+            registerListeners(pop);
+            if (showOverlay) {
+                addLoadingOverlay();
+            }
+            executeIfExists(crossmintOpened);
+        } else {
+            setConnecting(false);
+            console.log("Failed to open popup window");
```

Comment:
Can we make this a console.error and have it be more actionable for the developer? Can they do anything to fix this

Reply:
sure! @mPaella anything comes to mind?

Reply:
If it's not obvious let's leave a TODO and do in another task